src/os/bluestore/BlueStore.cc

```diff
@@ -5257,6 +5257,12 @@ int BlueStore::_setup_block_symlink_or_file(
   if (create)
     flags |= O_CREAT;
   if (epath.length()) {
+    if (!create) {
+      struct stat st;
+      r = ::stat(epath.c_str(), &st);
```

Comment:
could just use `std::experimental::filesystem::exists()`, in spite of the `std::experimental` naming space, it's in ISO C++17 standard. libstdc++ shipped along with GCC-7 still keep it in `std::experimental`, but GCC-8 gets this fixed, see https://github.com/gcc-mirror/gcc/commit/3b90ed62fb848046ed7ddef07df7c806e7f3fadb .